pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -177,7 +177,17 @@ func mergeService(dst, src *model.Service, srcRegistry serviceregistry.Instance)
 	// prefer the k8s VIP where possible
 	clusterID := srcRegistry.Cluster()
 	if srcRegistry.Provider() == provider.Kubernetes || len(dst.ClusterLocal.ClusterVIPs.GetAddressesFor(clusterID)) == 0 {
-		dst.ClusterLocal.ClusterVIPs.SetAddressesFor(clusterID, []string{src.Address})
+		dst.ClusterLocal.ClusterVIPs.SetAddressesFor(clusterID, src.ClusterLocal.ClusterVIPs.GetAddressesFor(clusterID))
+	}
+	if srcRegistry.Provider() == provider.Kubernetes {
+		// Copy over Kubernetes Multi-Cluster Services (MCS) ClusterSet Host and IPs.
+		if len(dst.ClusterSetLocal.Hostname) == 0 {
+			dst.ClusterSetLocal.Hostname = src.ClusterSetLocal.Hostname
+		}
+		if len(dst.ClusterSetLocal.ClusterVIPs.GetAddressesFor(clusterID)) == 0 {
```

Comment:
I am not sure I understand how this can be false or why we need it?

Reply:
I guess this is copied from cluster local logic, but the logic is provider = k8s || address is enpty there.

Reply:
Yeah that's right. Dst starts out empty and we just add to it. So the logic is effectively, if it hasn't been written to, write to it ... or if it's coming from a k8s cluster, always overwrite. The comment at the top I think is the key ... we always prefer vips from k8s.